### PR TITLE
NAS-120049 / 22.12.1 / Adding sudo fields to replication and connection forms

### PR DESCRIPTION
**Pull request by undsoft**
Adds new field to Advanced replication form and to an SSH connection form in automatic mode, when username is not root.
Testing requires latest Bluefin.

### CODE REVIEW

**Comment by undsoft** [comment]
@themylogin Please take a look at wording. I also didn't add this checkbox to a replication wizard. Not sure if we need it there, feels too technical. Perhaps we can make it easier on the user somehow.

File: src/app/helptext/data-protection/replication/replication-wizard.ts [hunk 1]
```diff
@@ -48,6 +48,9 @@ export default {
  <code>/zvol1</code> after <i>dataset1</i> will create <i>zvol1</i> for \
  snapshot storage.'),
 
+  sudo_tooltip: T('Controls whether the user used for SSH/SSH+NETCAT replication should use sudo to execute zfs commands on the remote host.\
```
Comment (themylogin) [inline src/app/helptext/data-protection/replication/replication-wizard.ts:51]:
```suggestion
  sudo_tooltip: T('Controls whether the user used for SSH/SSH+NETCAT replication will have passwordless sudo enabled to execute zfs commands on the remote host.\
```

File: src/app/pages/credentials/backup-credentials/ssh-connection-form/ssh-connection-form.component.html [hunk 2]
```diff
@@ -81,6 +81,13 @@
           [tooltip]="helptext.username_tooltip | translate"
         ></ix-input>
 
+        <ix-checkbox
+          *ngIf="!isManualSetup && (isNotRootUsername$ | async)"
+          formControlName="sudo"
+          [label]="'Allow sudo zfs commands with no password' | translate"
```
Comment (themylogin) [inline src/app/pages/credentials/backup-credentials/ssh-connection-form/ssh-connection-form.component.html:87]:
```suggestion
          [label]="'Enable passwordless sudo for zfs commands' | translate"
```

**Review by themylogin — CHANGES_REQUESTED** [review]

**Review by denysbutenko — APPROVED** [review]
image

image

Wonder if zfs needs to be uppercased. It looks good.

**Pull request merged**